embassy-stm32/src/eth/generic_smi.rs

```diff
@@ -41,21 +41,18 @@ mod phy_consts {
 }
 use self::phy_consts::*;
 
+#[non_exhaustive]
 /// Generic SMI Ethernet PHY
 pub struct GenericSMI {
     #[cfg(feature = "time")]
     poll_interval: Duration,
-    #[cfg(not(feature = "time"))]
```

Comment:
this is needed so the struct doesn't become publically-constructible with `GenericSMI{}` if `time` is disabled.